pkg/common/common.go

```diff
@@ -0,0 +1,35 @@
+package common
+
+import (
+	"regexp"
+
+	"github.com/jykuo-love-shiritori/twp/db"
+	"github.com/jykuo-love-shiritori/twp/pkg/constants"
+)
+
+type QueryParams struct {
+	Offset int32 `query:"offset"` // the start index of the query
+	Limit  int32 `query:"limit"`  // the number of the query
+}
+
+func NewQueryParams(offset int32, limit int32) QueryParams {
+	return QueryParams{Offset: offset, Limit: limit}
+}
+
+func (q *QueryParams) Validate() bool {
+	if q.Offset < 0 || q.Limit < 0 || q.Limit > constants.QUERY_LIMIT {
+		return false
+	}
+	return true
+}
+
+type Cart struct {
+	Seller_name string
+	Products    []db.GetProductInCartRow
+}
```

Comment:
only the seller.go use this struct

Reply:
The structure issue would not fix in this PR, it would be at the `router/buyer/common.go` in the future.